shaal/DrupalPod
# Simplify running drush, composer and yarn commands

# The Problem/Issue/Bug
DrupalPod tries keeping custom image as small as possible, so it only includes ddev binary.
You can run `drush`, `composer`, and `yarn` commands in ddev, but it creates some friction in the user experience.

## How this PR Solves The Problem
This PR makes `drush`, `composer`, and `yarn` commands that can run from host, but will actually run inside ddev container, in the current corresponding directory.

Edit: This PR also adds `node`, `nvm`, `npx` as programs that can be called from host, but run in the container.
`php` script was improved to use `ddev exec_d` instead of just `ddev exec`

fixes https://github.com/shaal/DrupalPod/issues/82

## Manual Testing Instructions
1. Open this PR in Gitpod - https://gitpod.io/#https://github.com/shaal/DrupalPod/pull/83
1. Run `composer update`
1. Confirm composer updated packages.
1. Run `drush cr`
1. Confirm drush run as expected, and cache was refreshed.
1. Change directory - `cd web/core`
1. Run `yarn install`
1. Confirm yarn is installing core's Node dependencies.

## Related Issue Link(s)

## Release/Deployment notes
